### PR TITLE
JENKINS-26097: Switch to new label expression validation API

### PR DESCRIPTION
**Pull request by Zastai**
This also incrementalifies the plugin.

Looks like there's not currently scope for adding auto-completion to LabelExpAxis, except by reworking the UX from a textarea to a set of textboxes.

### CODE REVIEW

**Comment by jglick** [comment]
Amends #10 I think.

File: .mvn/extensions.xml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<extensions xmlns="http://maven.apache.org/EXTENSIONS/1.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/EXTENSIONS/1.0.0 http://maven.apache.org/xsd/core-extensions-1.0.0.xsd">
+  <extension>
+    <groupId>io.jenkins.tools.incrementals</groupId>
+    <artifactId>git-changelist-maven-extension</artifactId>
+    <version>1.1</version>
```
Comment (jglick) [inline .mvn/extensions.xml:5]: Doing a newer version in #67 BTW.

File: pom.xml [hunk 2]
```diff
@@ -7,13 +7,15 @@
...
+    <version>${revision}${changelist}</version>
     <packaging>hpi</packaging>
     <name>Matrix Project Plugin</name>
     <description>Multi-configuration (matrix) project type.</description>
     <url>https://github.com/jenkinsci/matrix-project-plugin</url>
     <properties>
-      <jenkins.version>2.164.3</jenkins.version>
+      <revision>1.17</revision>
+      <changelist>-SNAPSHOT</changelist>
+      <jenkins.version>2.243</jenkins.version>
```
Comment (jglick) [inline pom.xml:18]: 2.2
```suggestion
      <jenkins.version>2.249.1</jenkins.version>
```

File: pom.xml [hunk 3]
```diff
@@ -70,6 +70,13 @@
             <artifactId>sshd</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <!-- Without this, org.jenkins-ci.modules:sshd:2.6 pulls in the too-old 1.4. -->
```
Comment (jglick) [inline pom.xml:74]: Could be dropped from this PR I think?

Comment (Zastai) [reply pom.xml:74]: Well without this, `mvn compile` fails, so the PR can't be validated.

```
[WARNING] Rule 5: org.apache.maven.plugins.enforcer.RequireUpperBoundDeps failed with message:
Failed while enforcing RequireUpperBoundDeps. The error(s) are [
Require upper bound dependencies error for org.jenkins-ci.modules:ssh-cli-auth:1.4 paths to dependency are:
+-org.jenkins-ci.plugins:matrix-project:1.17-SNAPSHOT
  +-org.jenkins-ci.modules:sshd:2.6
    +-org.jenkins-ci.modules:ssh-cli-auth:1.4
and
+-org.jenkins-ci.plugins:matrix-project:1.17-SNAPSHOT
  +-org.jenkins-ci.main:jenkins-war:2.243
    +-org.jenkins-ci.modules:ssh-cli-auth:1.8
```

Comment (jglick) [reply pom.xml:74]: Still? You are no longer changing anything else in the POM in this PR, after I merged in #68.

Comment (Zastai) [reply pom.xml:74]: Never mind - had pulled master, but not the branch. With HEAD, it can indeed be dropped.

**Review by jglick — APPROVED** [review]

**Pull request merged**